TSPL.docc/ReferenceManual/Expressions.md

```diff
@@ -921,6 +921,93 @@ to make use of the implementation in their superclass.
 >
 > *superclass-initializer-expression* → **`super`** **`.`** **`init`**
 
+### Conditional Expression
+
+A *conditional expression* evaluates to one of several given values
+based on the value of a condition.
+It has one the following forms:
+
+```swift
+if <#condition 1#> {
+   <#expression used if condition 1 is true#>
+} else if <#condition 2#> {
+   <#expression used if condition 2 is true#>
+} else {
+   <#expression used if both conditions are false#>
+}
+
+switch <#expression#> {
+case <#pattern 1#>:
+    <#expression 1#>
+case <#pattern 2#> where <#condition#>:
+    <#expression 2#>
+default:
+    <#expression 3#>
+}
+```
+
+A conditional expression
+has the same behavior and syntax as an `if` statement or a `switch` statement,
+except for the differences listed below.
+
+A conditional expression appears only in the following contexts:
+
+  - As the value assigned to a variable.
+  - As the initial value in a variable or constant declaration.
+  - As the error thrown by a `throw` expression.
+  - As the value returned by a function, closure, or property getter.
+
+The branches of a conditional expression are exhaustive,
+ensuring that the expression always produces a value
+regardless of the condition.
+This means each `if` branch needs a corresponding `else` branch.
+
+Each branch contains either a single expression,
+which is used as the value for the conditional expression
+when that branch's conditional is true,
+a `throw` statement,
+or a call to a function that never returns.
+
+Each branch must produce a value of the same type.
+Because type checking of each branch is independent,
+conditional expressions that that include a `nil` literal
+or different kinds of literals typically need type context,
+like an explicit type annotation.
+
+```swift
+let number: Double = if someCondition { 10 } else {12.34 }
+```
```

Comment:
Do we think this scenario is common enough that the guide should include a long-form example, walking folks through why the code doesn't work without a type annotation and then adding the type annotation?

Reply:
I'm hoping it won't come up _too_ often, though I think `nil` in particular is likely to be a not-uncommon case, so it might be worth adding a long-form example.

Reply:
Added an example to the guide in 4c5bc888.